paddle/framework/operator.h

```diff
@@ -317,26 +318,108 @@ class ExecutionContext : public InferShapeContext {
   const platform::DeviceContext& device_context_;
 };
 
+class CompileTimeInferShapeContext : public InferShapeContextBase {
+ public:
+  CompileTimeInferShapeContext(const OpDescBind& op, const BlockDescBind& block)
+      : op_(op), block_(block) {}
+
+  bool HasInput(const std::string& name) const override {
+    const std::vector<std::string>& input_names = op_.Input(name);
+    PADDLE_ENFORCE_EQ(input_names.size(), 1UL, "Inputs(%s) length is not 1",
```

Comment:
the
> Inputs(%s) length is not 1

lacks information, tell why should be 1

Reply:
I second @Superjom that this error message could be more informative. Why must the input length be 1? What is the "input" here?

Reply:
done